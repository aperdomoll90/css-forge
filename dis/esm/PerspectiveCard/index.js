export { PerspectiveCard } from "./PerspectiveCard";
export * from './PerspectiveCard.types';
//# sourceMappingURL=index.js.map