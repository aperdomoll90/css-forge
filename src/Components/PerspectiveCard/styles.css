:root {
  --bubble-card-background: #1c2942;
  --bubble-card-border: #67df6f47;
  --bubble-card-bubble: #67df6e;
  --bubble-card-shadow: #00000033;
  --bubble-card-background-letters: #bbbbbb0b;
  --bubble-card-white: #f2f2f2;
  /* --secondary-color: #161d26;
  --magic-mint: #b5ffe1;
  --violet-color: #541388;
  --gamboge: #e59500;
  --amaranth: #d93654;
  --transparent-metal: #15151596;
  --metal: #151515; */
}
.perspectiveCard-wrapper {
  width: 17rem;
  height: 23rem;
  position: relative;
  display: flex;
  align-items: center;
  justify-self: center;
  transform-style: preserve-3d;
}

/* .perspectiveCard-wrapper::before {
  content: '';
  width: 17rem;
  height: 15rem;
  border-radius: 1rem;
  position: absolute;
  top: 57%;
  transition: 0.5s ease-in-out;
  animation-delay: .2s;
  background: linear-gradient(to top, rgba(0, 0, 0, 0.174), transparent);
  transform-style: preserve-3d;
  filter: blur(15px);
  transform: perspective(5000px) translateX(-1.9rem) rotateY(-20deg) rotateX(-75deg);
} */

/* .perspectiveCard-wrapper:hover::before {
  transform: translateY(-5rem) translateX(1.9rem);
  opacity: 0;
} */
.perspectiveCard-content {
  width: 17rem;
  height: 23rem;
  position: absolute;
  top: 0;
  left: 0;
  border: 0.1rem solid transparent;
  box-shadow: 0 0.2rem 0.6rem var(--bubble-card-shadow);
  background: var(--bubble-card-background);
  border-radius: 1rem;
  z-index: 11;
  overflow: hidden;
  /* transition: 0.5s ease-in-out; */
  -webkit-box-reflect: below 5px linear-gradient(transparent, transparent, #0004);
  transform-style: preserve-3d;
  transform: perspective(5000px) rotateY(var(--rotateY)) rotateX(var(--rotateX));
}

.perspectiveCard-content:hover {
  border: 0.1rem solid var(--bubble-card-border);
  /* transform: scale(1.05); */
}
.perspectiveCard-content::before,
.perspectiveCard-content::after {
  content: '';
  position: absolute;
  z-index: 5;
  background: var(--bubble-card-bubble);
  border-radius: 50%;
  transition: 0.5s ease-in-out;
}

.perspectiveCard-content::before {
  width: 9rem;
  height: 9rem;
  top: -6%;
  right: -6%;
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover::before {
  transform: scale(1.8);
}

.perspectiveCard-content::after {
  width: 5rem;
  height: 5rem;
  bottom: -6%;
  left: -6%;
}

.perspectiveCard-content:hover::after {
  transform: scale(0);
}

.perspectiveCard-marquee {
  position: absolute;
  top: 40%;
  left: -15%;
  z-index: 0;
  font-size: 11rem;
  font-weight: 900;
  text-transform: uppercase;
  font-family: 'Anton', sans-serif;
  color: var(--bubble-card-background-letters);
  text-shadow: 0 0.2rem 2rem var(--bubble-card-shadow);
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover .perspectiveCard-marquee {
  transform: translateX(-15rem);
  /* opacity: 0; */
}
.perspectiveCard-img {
  width: 70%;
  height: auto;
  position: absolute;
  bottom: 25%;
  left: 15%;
  z-index: 20;
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover .perspectiveCard-img {
  transform: scale(0.8) translateX(-3rem) translateY(-5rem);
}

.perspectiveCard-title {
  font-size: 1.8rem;
  letter-spacing: 0.2rem;
  position: absolute;
  bottom: 13%;
  right: 25%;
  text-shadow: 0 0 0.5rem #00000085;
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover .perspectiveCard-title {
  transform: scale(0.8) translateX(4rem) translateY(-6rem);
}

.perspectiveCard-body {
  width: 100%;
  height: 10rem;
  position: relative;
  bottom: -100%;
  left: 0;
  z-index: 15;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: 0.9s ease-in-out;
  animation-delay: 0.2s;
}
.perspectiveCard-content-text {
  margin: 0.5rem 1.5rem;
  position: absolute;
  top: 1rem;
  font-weight: 400;
}
.perspectiveCard-content:hover .perspectiveCard-body {
  transform: translateY(-100%);
}
.perspectiveCard-body p,
h1 {
  font-family: sans-serif;
  color: var(--bubble-card-white);
}

.perspectiveCard-footer {
  width: 100%;
  height: 3rem;
  position: absolute;
  bottom: -3.5rem;
  left: 0;
  z-index: 20;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  transition: 0.9s ease-in-out;
  animation-delay: 0.3s;
}

.perspectiveCard-content:hover .perspectiveCard-footer {
  transform: translateY(-150%);
}
.perspectiveCard-btn {
  padding: 1rem 2rem;
  background-color: unset;
  border: none;
  outline: none;
  border-radius: 2rem;
  color: var(--bubble-card-white);
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
}
.perspectiveCard-success {
  background-color: var(--bubble-card-bubble);
}
.perspectiveCard-border {
  border: 0.1rem solid var(--bubble-card-bubble);
}
