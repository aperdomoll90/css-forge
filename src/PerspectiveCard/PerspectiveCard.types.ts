export interface PerspectiveCardPropsType {
    label: string
    content: string
    imgUrl: string
    demoUrl: string
    codeUrl: string
  }