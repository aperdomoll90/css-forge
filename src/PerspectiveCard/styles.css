:root {
  --bubble-card-background: #1c2942;
  --bubble-card-border: #67df6f47;
  --bubble-card-bubble: #67df6e;
  --bubble-card-button-active: #2c5d30;
  --bubble-card-shadow: #00000033;
  --bubble-card-background-letters: #bbbbbb0b;
  --bubble-card-white: #f2f2f2;
}
.perspectiveCard-wrapper {
  width: var(--width);
  height: var(--height);
  position: relative;
  display: flex;
  align-items: center;
  justify-self: center;
  transform-style: preserve-3d;
}

.perspectiveCard-content {
  width: var(--width);
  height: var(--height);
  position: absolute;
  top: 0;
  left: 0;
  border: 0.1rem solid transparent;
  box-shadow: 0 0.2rem 0.6rem var(--bubble-card-shadow);
  background: var(--primaryColor);
  border-radius: 1rem;
  z-index: 11;
  overflow: hidden;
  -webkit-box-reflect: below 5px linear-gradient(transparent, transparent, #0004);
  transform-style: preserve-3d;
  transform: perspective(5000px) rotateY(var(--rotateY)) rotateX(var(--rotateX));
}

.perspectiveCard-content:hover {
  border: 0.1rem solid var(--secondaryColor);
}
.perspectiveCard-content::before,
.perspectiveCard-content::after {
  content: '';
  position: absolute;
  z-index: 5;
  background: var(--secondaryColor);
  border-radius: 50%;
  transition: 0.5s ease-in-out;
}

.perspectiveCard-content::before {
  width: 9rem;
  height: 9rem;
  top: -6%;
  right: -6%;
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover::before {
  transform: scale(1.8);
}

.perspectiveCard-content::after {
  width: 5rem;
  height: 5rem;
  bottom: -6%;
  left: -6%;
}

.perspectiveCard-content:hover::after {
  transform: scale(0);
}

.perspectiveCard-marquee {
  position: absolute;
  top: 40%;
  left: -15%;
  z-index: 0;
  font-size: 11rem;
  font-weight: 900;
  text-transform: uppercase;
  font-family: 'Anton', sans-serif;
  color: var(--bubble-card-background-letters);
  text-shadow: 0 0.2rem 2rem var(--bubble-card-shadow);
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover .perspectiveCard-marquee {
  transform: translateX(-15rem);
  /* opacity: 0; */
}
.perspectiveCard-img {
  width: var(--imgWidth);
  height: auto;
  position: absolute;
  bottom: 25%;
  left: 15%;
  z-index: 20;
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover .perspectiveCard-img {
  transform: scale(0.8) translateX(-3rem) translateY(-5rem);
}

.perspectiveCard-title {
  font-size: 1.8rem;
  letter-spacing: 0.2rem;
  position: absolute;
  bottom: 13%;
  right: 25%;
  text-shadow: 0 0 0.5rem #00000085;
  transition: 1s ease-in-out;
}

.perspectiveCard-content:hover .perspectiveCard-title {
  transform: scale(0.8) translateX(4rem) translateY(-6rem);
}

.perspectiveCard-body {
  width: 100%;
  height: 10rem;
  position: relative;
  bottom: -100%;
  left: 0;
  z-index: 15;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: 0.9s ease-in-out;
  animation-delay: 0.2s;
}
.perspectiveCard-content-text {
  margin: 0.5rem 1.5rem;
  position: absolute;
  top: 1rem;
  font-weight: 400;
}
.perspectiveCard-content:hover .perspectiveCard-body {
  transform: translateY(-100%);
}
.perspectiveCard-body p,
h1 {
  font-family: sans-serif;
  color: var(--fontColor);
}

.perspectiveCard-footer {
  width: 100%;
  height: 3rem;
  position: absolute;
  bottom: -3.5rem;
  left: 0;
  z-index: 20;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  transition: 0.9s ease-in-out;
  animation-delay: 0.3s;
}

.perspectiveCard-content:hover .perspectiveCard-footer {
  transform: translateY(-150%);
}
.perspectiveCard-btn {
  padding: 1rem 2rem;
  background-color: unset;
  outline: none;
  border: none;
  border-radius: 2rem;
  color: var(--fontColor);
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: 0.5s ease-in-out;
}
.perspectiveCard-btn:hover {
  background-color: var(--buttonActive);
  outline: 0.1rem solid var(--buttonActive);
}

.perspectiveCard-success {
  background-color: var(--secondaryColor);
  outline: 0.1rem solid var(--secondaryColor);
}
.perspectiveCard-border {
  outline: 0.1rem solid var(--secondaryColor);
}

.perspectiveCard-link {
  text-decoration: none;
}
