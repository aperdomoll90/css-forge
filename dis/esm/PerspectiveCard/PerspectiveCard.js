var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect } from 'react';
import { rotateElement } from '../utils/MouseEvents';
import './styles.css';
import '../utils/GlobalStyles.css';
import { LightenDarkenColor } from '../utils/ColorManipulation';
export var PerspectiveCard = function (_a) {
    var width = _a.width, height = _a.height, primaryColor = _a.primaryColor, secondaryColor = _a.secondaryColor, fontColor = _a.fontColor, label = _a.label, content = _a.content, imgUrl = _a.imgUrl, imgWidth = _a.imgWidth, demoUrl = _a.demoUrl, codeUrl = _a.codeUrl;
    useEffect(function () {
        var contentArea = document.querySelector('.perspectiveCard-content');
        document.addEventListener('mousemove', function (e) {
            rotateElement(e, contentArea);
        });
    }, []);
    var stylesProps = {
        '--width': width ? "".concat(width, "rem") : '17rem',
        '--height': height ? "".concat(height, "rem") : '23rem',
        '--imgWidth': imgWidth ? "".concat(imgWidth, "%") : '70%',
        '--primaryColor': primaryColor ? primaryColor : '#1c2942',
        '--secondaryColor': secondaryColor ? secondaryColor : '#67df6e',
        '--fontColor': fontColor ? fontColor : '#fff',
        '--buttonActive': secondaryColor ? LightenDarkenColor(secondaryColor, -50) : LightenDarkenColor('#67df6e', -50),
    };
    return (_jsx("div", __assign({ className: 'perspectiveCard-wrapper', style: stylesProps }, { children: _jsxs("div", __assign({ className: 'perspectiveCard-content' }, { children: [_jsx("div", __assign({ className: 'perspectiveCard-marquee' }, { children: label })), _jsx("div", __assign({ className: 'perspectiveCard-img-container' }, { children: _jsx("img", { alt: "".concat(label, "-image"), src: imgUrl, className: 'perspectiveCard-img' }) })), _jsx("h1", __assign({ className: 'perspectiveCard-title' }, { children: label })), _jsx("div", __assign({ className: 'perspectiveCard-body' }, { children: _jsx("p", __assign({ className: 'perspectiveCard-content-text' }, { children: content })) })), _jsxs("div", __assign({ className: 'perspectiveCard-footer' }, { children: [demoUrl && (_jsx("a", __assign({ className: 'perspectiveCard-link', target: '_blank', href: demoUrl }, { children: _jsx("button", __assign({ className: 'perspectiveCard-btn perspectiveCard-success' }, { children: "Demo" })) }))), codeUrl && (_jsx("a", __assign({ className: 'perspectiveCard-link', target: '_blank', href: codeUrl }, { children: _jsx("button", __assign({ className: 'perspectiveCard-btn perspectiveCard-border' }, { children: "Github" })) })))] }))] })) })));
};
//# sourceMappingURL=PerspectiveCard.js.map