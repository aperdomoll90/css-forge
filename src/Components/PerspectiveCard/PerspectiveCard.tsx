import { useEffect } from 'react'
import './styles.css'
import { rotateElement } from '../BubbleCard/utils/MouseEvents'

export interface PerspectiveCardPropsType {
  color?: string
  label: string
  content: string
  imgUrl: string
  linkUrl?: string
}

function PerspectiveCard({ color, label, content, imgUrl, linkUrl }: PerspectiveCardPropsType) {
  
  useEffect(() => {
    const contentArea = document.querySelector('.perspectiveCard-content')

    document.addEventListener('mousemove', e => {
      rotateElement(e, contentArea)
    })
  
  }, [])
  return (
    <div className='perspectiveCard-wrapper'>
      <div className='perspectiveCard-content'>
        <div className='perspectiveCard-marquee'>{label}</div>
        <img alt={`${label}-image`} src={imgUrl} className='perspectiveCard-img' />
        <h1 className='perspectiveCard-title'>{label}</h1>
        <div className='perspectiveCard-body'>
          <p className='perspectiveCard-content-text'>{content}</p>
        </div>
        <div className='perspectiveCard-footer'>
          <button className='perspectiveCard-btn perspectiveCard-success'>Demo</button>
          <button className='perspectiveCard-btn perspectiveCard-border'>Github</button>
        </div>
      </div>
    </div>
  )
}

export default PerspectiveCard
