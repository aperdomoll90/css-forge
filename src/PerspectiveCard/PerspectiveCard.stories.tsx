import { Meta, StoryFn } from '@storybook/react'
import { PerspectiveCardPropsType } from './PerspectiveCard.types'
import { PerspectiveCard } from './PerspectiveCard'
import StorybookContainer from '../StorybookContainer'
import shoe from './media/shoe.png'

export default {
  title: 'Cards',
  component: PerspectiveCard,
  parameters: {
    jest: ['PerspectiveCard.test.tsx'],
  },
  argTypes: {
    color: { options: ['red', ''], control: { type: 'radio' } },
    label: { options: ['Sea', 'Forest', 'Mountain'], control: { type: 'radio' } },
  },
} as Meta

const PerspectiveCardTemplate: StoryFn<PerspectiveCardPropsType> = args => (
  <StorybookContainer>
    <PerspectiveCard {...args} />
  </StorybookContainer>
)

export const PerspectiveCardComponent = PerspectiveCardTemplate.bind({})
PerspectiveCardComponent.args = {
  label: 'Shoes',
  imgUrl: shoe,
  content: 'This card follows the pointer',
  demoUrl: 'https://ingots.web.app/fishing',
  codeUrl: 'https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll',
}
