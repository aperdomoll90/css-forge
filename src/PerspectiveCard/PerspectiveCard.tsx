import { useEffect } from 'react'
import { rotateElement } from '../utils/MouseEvents'
import { PerspectiveCardPropsType } from './PerspectiveCard.types'
import './styles.css'

export const PerspectiveCard: React.FC<PerspectiveCardPropsType> = ({ label, content, imgUrl, demoUrl, codeUrl }) => {
  useEffect(() => {
    const contentArea: HTMLDivElement | null = document.querySelector('.perspectiveCard-content')

    document.addEventListener('mousemove', e => {
      rotateElement(e, contentArea)
    })
  }, [])
  return (
    <div className='perspectiveCard-wrapper'>
      <div className='perspectiveCard-content'>
        <div className='perspectiveCard-marquee'>{label}</div>
        <img alt={`${label}-image`} src={imgUrl} className='perspectiveCard-img' />
        <h1 className='perspectiveCard-title'>{label}</h1>
        <div className='perspectiveCard-body'>
          <p className='perspectiveCard-content-text'>{content}</p>
        </div>
        <div className='perspectiveCard-footer'>
          <a className='perspectiveCard-link' target='_blank' href={demoUrl}>
            <button className='perspectiveCard-btn perspectiveCard-success'>Demo</button>
          </a>
          <a className='perspectiveCard-link' target='_blank' href={codeUrl}>
            <button className='perspectiveCard-btn perspectiveCard-border'>Github</button>
          </a>
        </div>
      </div>
    </div>
  )
}
