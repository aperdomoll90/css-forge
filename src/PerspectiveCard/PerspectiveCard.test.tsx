import { render } from '@testing-library/react'
import { PerspectiveCard } from './PerspectiveCard'
import shoe from './media/shoe.png'
import { PerspectiveCardPropsType } from './PerspectiveCard.types'

describe('PerspectiveCard component', () => {
  // const { axe, toHaveNoViolations } = require('jest-axe')
  // expect.extend(toHaveNoViolations)
  const props:PerspectiveCardPropsType = {
    label: 'label',
    imgUrl: shoe,
    content: 'This card follows the pointer',
    demoUrl: 'https://ingots.web.app/fishing',
    codeUrl: 'https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll',
  }

  const renderComponent = () => render(<PerspectiveCard {...props} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
  //   it('should demonstrate ADA compliance on this component', async () => {
  //     const { container } = renderComponent()
  //     const results = await axe(container)
  //     expect(container).toHaveNoViolations()
  //   })
})
