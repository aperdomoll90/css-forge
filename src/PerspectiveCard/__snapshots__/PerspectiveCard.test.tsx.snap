// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerspectiveCard component should match snapshots 1`] = `
<div>
  <div
    class="perspectiveCard-wrapper"
  >
    <div
      class="perspectiveCard-content"
    >
      <div
        class="perspectiveCard-marquee"
      >
        label
      </div>
      <img
        alt="label-image"
        class="perspectiveCard-img"
        src="shoe.png"
      />
      <h1
        class="perspectiveCard-title"
      >
        label
      </h1>
      <div
        class="perspectiveCard-body"
      >
        <p
          class="perspectiveCard-content-text"
        >
          This card follows the pointer
        </p>
      </div>
      <div
        class="perspectiveCard-footer"
      >
        <a
          class="perspectiveCard-link"
          href="https://ingots.web.app/fishing"
          target="_blank"
        >
          <button
            class="perspectiveCard-btn perspectiveCard-success"
          >
            Demo
          </button>
        </a>
        <a
          class="perspectiveCard-link"
          href="https://github.com/aperdomoll90/ingots/tree/main/src/Archives/SvgPathOnScroll"
          target="_blank"
        >
          <button
            class="perspectiveCard-btn perspectiveCard-border"
          >
            Github
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
